test/net/sourceforge/kolmafia/preferences/PreferencesTest.java

```diff
@@ -2,25 +2,59 @@
 
 import static org.junit.jupiter.api.Assertions.*;
 
+import java.io.File;
 import java.util.TreeMap;
 import net.sourceforge.kolmafia.KoLCharacter;
-import org.junit.jupiter.api.*;
+import net.sourceforge.kolmafia.KoLmafia;
+import net.sourceforge.kolmafia.KoLmafiaCLI;
+import net.sourceforge.kolmafia.textui.command.AbstractCommand;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 
 class PreferencesTest {
-
-  // Convert to the new way of doing things...
-  @BeforeAll
-  protected static void initAll() {
+  @BeforeEach
+  public void initializeCharPrefs() {
     KoLCharacter.reset("fakePrefUser");
     KoLCharacter.reset(true);
     Preferences.saveSettingsToFile = false;
   }
 
+  @AfterEach
+  public void removePrefs() {
+    deleteUserPrefsAndMoodsFiles(KoLCharacter.baseUserName());
+    deleteGlobals();
+    KoLCharacter.reset("");
+    KoLCharacter.reset(true);
+    KoLCharacter.setUserId(0);
+    Preferences.saveSettingsToFile = false;
+    AbstractCommand.clear();
+    KoLmafiaCLI.registerCommands();
+    KoLmafia.lastMessage = "";
+    KoLmafia.forceContinue();
+  }
+
+  public static void deleteUserPrefsAndMoodsFiles(String user) {
+    String begin = "settings/" + user;
+    File file = new File(begin + "_prefs.txt");
+    if (file.exists()) {
+      file.deleteOnExit();
```

Comment:
Do you want to delete this only when the VM terminates? This reads like it has the potential to leak state between tests.

Reply:
I watched this and no tests fail and test\root\sessions was empty at the end so it did what I wanted.  file.deleteOnExit() is a defensive move from a multi-threaded environment where file.delete() executed immediately and caused errors on other threads.  I'd like to understand more about how JUnit is threading before I change to file.delete() since everything I can observe at the OS and debugger level says this is working and there are no leaks.